src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkRepositoryModule.java

```diff
@@ -107,11 +107,19 @@ public StarlarkCallable repositoryRule(
       }
     }
     builder.setConfiguredTargetFunction(implementation);
-    // TODO(b/291752414): If we care about the digest of repository rules, we should be using the
-    // transitive bzl digest of the module of the outermost stack frame, not the innermost.
-    BazelModuleContext moduleContext = BazelModuleContext.ofInnermostBzlOrThrow(thread);
-    builder.setRuleDefinitionEnvironmentLabelAndDigest(
-        moduleContext.label(), moduleContext.bzlTransitiveDigest());
+    var bzlInitContext = (BzlInitThreadContext) thread.getThreadLocal(BazelStarlarkContext.class);
```

Comment:
This cast looks unsafe/wrong and also features in the CI error.

Reply:
indeed -- fixed